homeassistant/components/purpleair/config_flow.py

```diff
@@ -46,6 +46,13 @@
     }
 )
 
+SENSOR_INDEX_SCHEMA = vol.Schema(
+    {
+        # The PurpleAir sensor index is an integer, but the UI for a string fits better:
+        vol.Required(CONF_SENSOR_INDEX): cv.string,
```

Comment:
Probably better to use NumberSelector which gives you a float and then you can easily convert to int in your validation